src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/AzureUtils.java

```diff
@@ -169,6 +171,37 @@ public static SharedAccessBlobPolicy generateBlobPolicy() {
         return policy;
     }
 
+    public static String generateBlobSASURL(
+            StorageAccountInfo storageAccount,
+            String containerName,
+            String blobName,
+            EnumSet<SharedAccessBlobPermissions> permissions) throws Exception {
+
+        CloudStorageAccount cloudStorageAccount = getCloudStorageAccount(storageAccount);
+
+        // Create the blob client.
+        CloudBlobClient blobClient = cloudStorageAccount.createCloudBlobClient();
+        CloudBlobContainer container = blobClient.getContainerReference(containerName);
+
+        // At this point need to throw an error back since container itself did not exist.
+        if (!container.exists()) {
+            throw new Exception("WAStorageClient: generateBlobSASURL: Container " + containerName
```

Comment:
why not automatically create blob container?

Reply:
Indeed in the scanning step , it will create the container if it is not existing. Here is just a prevention action, rare case will hit this code like container is deleted after the scanning step.

Reply:
this is a public method, it should behave integrity of itself, no idea of upstream-downstream logic.

Reply:
Both uploading and downloading will use this method to generate SAS. These two operations can be used separately. For downloading, it makes no sense to create the container here.